crates/shared/src/sources/uniswap_v3/pool_fetching.rs

```diff
@@ -160,90 +168,228 @@ impl UniswapV3PoolFetcher {
 
         match pool_ids.peek() {
             Some(_) => {
-                let now = Instant::now();
-                let mut cache = self.cache.lock().unwrap();
-                pool_ids.partition_map(|pool_id| match cache.get_mut(pool_id) {
-                    Some(entry)
-                        if now.saturating_duration_since(entry.updated_at) < self.max_age =>
-                    {
-                        tracing::debug!("returning cached pool: {:?}", pool_id);
-                        entry.requested_at = now;
-                        Either::Left(entry.pool.clone())
+                let pools_checkpoint = self.pools_checkpoint.lock().unwrap();
+                pool_ids.partition_map(|pool_id| match pools_checkpoint.data.get(pool_id) {
+                    Some(entry) => {
+                        tracing::debug!("returning pool: {:?}", pool_id);
```

Comment:
Judging by the use of `peekable()` you want to avoid acquiring the lock if it is not strictly necessary. In that case I would recommend logging the pool ids after the lock has been released again to keep the critical section as short as possible.

Reply:
Removed log as not necessary